doc/user_guide/psyir.rst

```diff
@@ -579,3 +579,17 @@ would need to be set explicitly.
 
 This method is useful as it allows the class of a symbol to be changed
 without affecting any references to it.
+
+Replacing PSyIR nodes
+---------------------
+
+In certain cases one might want to replace a node in a PSyIR tree with
+another node. All nodes provide the `replace_with()` method to support
+this.
+
+.. code-block:: python
+
+    node.replace_with(new_node)
+
+Note, the node and all of its children is replaced with new_node and
```

Comment:
I think this is quite significant and should be part of the description above instead of just a note. Maybe:
"All nodes provide the `replace_with()` method to replace any specific node and all its children with another given node and all its children."

Reply:
Good point. Updated.